build/Targets/Analyzers.Settings.targets

```diff
@@ -31,29 +41,23 @@
     <ProjectLanguage Condition="'$(MSBuildProjectExtension)' == '.vbproj' OR '$(Language)' == 'VB'">VB</ProjectLanguage>
     <ProjectLanguage Condition="'$(MSBuildProjectExtension)' == '.vcxproj' OR '$(Language)' == 'C++'">C++</ProjectLanguage>
   </PropertyGroup>
-  
-  <!-- Import the global NuGet packages -->
-  <PropertyGroup>
-    <ToolsetCompilerPackageName>Microsoft.Net.Compilers</ToolsetCompilerPackageName>
-    <ToolsetCompilerPropsFilePath>$(NuGetPackageRoot)\Microsoft.Net.Compilers\2.0.0-rc4\build\Microsoft.Net.Compilers.props</ToolsetCompilerPropsFilePath>
+
+  <PropertyGroup Condition="'$(ProjectLanguage)' == 'CSharp' OR '$(ProjectLanguage)' == 'VB'">
+    <DotnetProject>true</DotnetProject>
   </PropertyGroup>
   
+  <!-- For C# and VB projects, this imports the .NET SDK which will pull in Microsoft.Common.props -->
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" Condition="'$(DotnetProject)' == 'true'" />
+
+  <!-- For non-managed and non-vcxproj projects, this imports Microsoft.Common.props -->
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props"
```

Comment:
Can't  you just get rid of the nuproj and call msbuild /t:pack instead?

---
In reply to: [108814528](https://github.com/dotnet/roslyn-analyzers/pull/1123#discussion_r108814528) [](ancestors = 108814528)